### DISCUSSION TITLE
Allowing objects to not have associated samples in specific examples

### DISCUSSION THREAD

**Discussion by bdach** [ppy/osu#25186] [Ideas]
When talking with the NAT about https://github.com/ppy/osu/pull/25185, the discussion touched on the point that there are specific cases where silent objects are desired:

> "Due to slider leniency you may have sliders extended so rather than letting go to a specific sound, you let go briefly to then head into the next slider. It's good for when the sound is pretty continuous with some hits during it"

>"Slider ends and spinner ends typically. Most commonly when they end on a blue tick or smaller division"
> "And those too when they go into a fade out"

> "F2'd a random map and this has a good example of it during the chorus 00:59:657 https://osu.ppy.sh/beatmapsets/1510572#osu/3093344"
> "This map uses 5% timing points to achieve it actually but the example is still there"

> "its also frequently used for slider-ticks aswell"

I know historically the RC has been pretty strict on having hitsounds, but maybe we can sort something out to support this better? Apparently mappers will still do this, just by a workaround of using silent samples, which bypass the 5%/8%/whatever volume floor, but do it via the use of a specific silent sound file, to the point where that specific file [is _also_ mentioned in the RC](https://osu.ppy.sh/wiki/pl/Ranking_criteria):

![image](https://github.com/ppy/osu/assets/20418176/25fab36a-3ecb-48c9-b33c-576a6281fa34)

I'm not opening this as a signal that this *needs* to be addressed, especially not that soon, but moreso to gauge the reaction to such a proposal. and discuss possible ways of making this easier.

**Comment by TheMoMan** [comment]
Just to add another example I had a case in a map I ranked where even 5% (or 8% whatever it was) was audible enough to mess with what I was going for https://osu.ppy.sh/beatmapsets/1280192#osu/2659351 01:51:412

**Reply by Okorin** [reply]
> I know historically the RC has been pretty strict on having hitsounds, but maybe we can sort something out to support this better? Apparently mappers will still do this, just by a workaround of using silent samples, which bypass the 5%/8%/whatever volume floor, but do it via the use of a specific silent sound file

not quite accurate:
- currently having hitsounds at all is optional for osu!mania
- osu! has this guideline:
  - **Spinner ends, slider ends, and slider reverses should have hitsound feedback.** If these are used to represent a held sound and do not align with a distinct sound, having no feedback is acceptable.
     - meaning that if the slider doesn't end on something that would qualify as a sound, right now the 44-byte file can be used if you want to have the slider end / spinner end or slider reverse be completely silent; I'd argue that 0% volume would be more intuitive for silencing something but the game currently doesn't allow it

If we had the ability to do this natively, we can get rid of that hack with the 44-byte sample, though i suppose it will stay for silencing all `sliderslide.wav`, `slidertick.wav` in a map (feedback on those is optional, in most situations)

I'd be all for this because it means we can simplify and don't need to do hacky-seeming workarounds 👍
